admin-dev/themes/new-theme/js/components/grid/extension/ajax-bulk-action-extension.ts

```diff
@@ -0,0 +1,193 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import {Grid} from '@PSTypes/grid';
+import ProgressModal from '@components/modal/progress-modal';
+import ConfirmModal from '@components/modal/confirm-modal';
+import GridMap from '@components/grid/grid-map';
+import Router from '@components/router';
+
+import ClickEvent = JQuery.ClickEvent;
+
+const {$} = window;
+
+/**
+ * Handles submit of grid actions
+ */
+export default class AjaxBulkActionExtension {
+  private router = new Router();
+
+  /**
+   * Extend grid with bulk action submitting
+   *
+   * @param {Grid} grid
+   */
+  extend(grid: Grid): void {
+    grid
+      .getContainer()
+      .on('click', GridMap.bulks.ajaxAction, (event: ClickEvent) => {
+        const $ajaxButton: JQuery<HTMLInputElement> = $<HTMLInputElement>(event.currentTarget);
+        const $checkboxes: JQuery<HTMLInputElement> = $<HTMLInputElement>(GridMap.bulks.checkedCheckbox);
+        const selectedIds: string[] = $checkboxes.get().map((checkbox: HTMLInputElement) => checkbox.value);
+
+        if (selectedIds.length === 0) {
+          return;
+        }
+
+        const confirmBulkAction = $ajaxButton.data('confirmBulkAction') ?? true;
+
+        if (confirmBulkAction) {
+          const progressionTitle = $ajaxButton.data('progressTitle');
+          const closeButtonLabel = $ajaxButton.data('cancelLabel') || 'Cancel';
+          const confirmTitle = $ajaxButton.data('confirmTitle') || 'Apply modifications';
+          const bulkAction = $ajaxButton.data('bulkAction') ?? 'bulk-action';
+
+          const confirmModal = new ConfirmModal({
+            id: GridMap.actions.ajaxBulkActionConfirmModal(grid.id, bulkAction),
+            modalTitle: confirmTitle,
+            closeButtonLabel,
+            confirmMessage: progressionTitle.replace('%total%', selectedIds.length),
+            confirmButtonLabel: confirmTitle,
+            confirmCallback: () => {
+              this.submitForm(grid, $<HTMLInputElement>(event.currentTarget), selectedIds);
+            },
+          });
+          confirmModal.show();
+        } else {
+          this.submitForm(grid, $<HTMLInputElement>(event.currentTarget), selectedIds);
+        }
+      });
+  }
+
+  private async submitForm(grid: Grid, $ajaxButton: JQuery<HTMLInputElement>, selectedIds: string[]): Promise<void> {
+    const bulkChunkSize = $ajaxButton.data('bulkChunkSize') ?? 10;
```

Comment:
Late comment (noticed some errors while applying this modal elsewhere):

as I see action has `data-request-bulk-chunk-size` so there is probably a bug (it should be requestBulkChunkSize)